src/ixdat/plotters/tpms_plotter.py

```diff
@@ -0,0 +1,606 @@
+            make_colorbar (bool): Whether to make a colorbar.
+                FIXME: colorbar at present mis-alignes axes
+            aspect (float): aspect ratio. Defaults to 1.25 times taller than wide.
+            max_threshold (float): Set maximum value in scanning data and set to zero if
+                data is above.
+            min_threshold (float): Set minimum value in scanning data and set to zero if
+                data is below.
+            scanning_mask (boolean list): List of booleans to include/ exclude specfic
+                data in scanning plot (specific masses that are monitored otherwise)
+            _sort_indicies (list): list of floats to sort data. Defaults low to high.
+            vmin (float): Value used to shift colours in the colorbar to lower values
+            vmax (float): Value to shift colours in the colorbar to higher values
+            kwargs (dict): Additional kwargs go to all calls of matplotlib's plot()
+
+        Returns:
+            list of Axes: (top, mid_left, bottom_left, mid_right, bottom_right) where:
+                axes[0] is top is MS spectral data
+                axes[1] is mid_left is Mass ID data;
+                axes[2] is bottom_left is temperature;
+                axes[3] is mid_right is additional MS data if left and right mass_lists
+                    or mol_lists were plotted (otherwise axes[3] is None); and
+                axes[4] is bottom_right is pressure.
+        """
+        measurement = measurement or self.measurement
+
+        if not axes:
+            axes = self.new_three_panel_axes(
+                n_top=1, n_middle=(2 if (mass_lists or mol_lists) else 1), n_bottom=2
+            )
+
+        measurement.spectrum_series.heat_plot(
+            ax=axes[0],
+            tspan=tspan,
+            xspan=xspan,
+            cmap_name=cmap_name,
+            make_colorbar=make_colorbar,
+            max_threshold=max_threshold,
+            min_threshold=min_threshold,
+            scanning_mask=scanning_mask,
+            _sort_indicies=_sort_indicies,
+            vmin=vmin,
+            vmax=vmax,
+        )
+
+        self.tpms_plotter.plot_measurement(
+            measurement=measurement,
+            axes=[axes[1], axes[2], axes[4], axes[5]],
+            tspan=tspan,
+            tspan_bg=tspan_bg,
+            remove_background=remove_background,
+            mass_list=mass_list,
+            mass_lists=mass_lists,
+            mol_list=mol_list,
+            mol_lists=mol_lists,
+            unit=unit,
+            logplot=logplot,
+            legend=legend,
+            T_name=T_name,
+            P_name=P_name,
+            T_color=T_color,
+            P_color=P_color,
+            **kwargs,
+        )
+
+        axes[0].set_xlim(axes[1].get_xlim())
+
+        fig = axes[0].get_figure()
+        fig.set_figheight(fig.get_figwidth() * aspect)
+
+        return axes
+
+    def plot_measurement_vs(
```

Comment:
This can easily be placed in the SpectroMSPlotter instead. I just wrote it here, because this is where I would use it and always vs T of P, hence the TPMSPlotter placement :-)